.pipelines/templates/template-prod-e2e-steps.yml

```diff
@@ -4,26 +4,46 @@ parameters:
   azureDevOpsE2EJSONSPN: ''
 
 steps:
-# TODO(mj) we need to checkout code to run tooling like kubeconfig. This should go away once we refactor everything into golang.
-- template: ./template-checkout.yml
-- template: ./template-az-cli-login.yml
-  parameters:
-    azureDevOpsJSONSPN: ${{ parameters.azureDevOpsE2EJSONSPN }}
 - script: |
     set -e
     export LOCATION=${{ parameters.location }}
     export AZURE_SUBSCRIPTION_ID=${{ parameters.subscription }}
 
+    # TODO: need to fetch $RP_IMAGE_ACR and $ARO_VERSION_SA, ideally from the RP
+    # config.  Would be good to have the original git commit of the RP config
+    # and use it.
+
     trap 'rm -f devops-spn.json' EXIT
     base64 -d >devops-spn.json <<<${{ parameters.azureDevOpsE2EJSONSPN }}
     export AZURE_CLIENT_ID=$(jq -r .clientId <devops-spn.json)
     export AZURE_CLIENT_SECRET=$(jq -r .clientSecret <devops-spn.json)
     export AZURE_TENANT_ID=$(jq -r .tenantId <devops-spn.json)
 
     set -x
-    . ./hack/e2e/run-rp-and-e2e.sh
 
-    export CI=true
-    run_e2e
+    az acr login -n $RP_IMAGE_ACR
```

Comment:
here, do docker login with CLIENT_ID and CLIENT_SECRET (much preferred than bringing back az, if this is feasible)

Reply:
and don't forget the docker logout: globally update the trap to `trap 'for c in $(docker ps -aq); do docker rm -f $c; done; docker image prune -af; docker logout -a' EXIT` and use it here as well